gtsfm/frontend/matcher/superglue_matcher.py

```diff
@@ -0,0 +1,108 @@
+"""SuperGlue matcher implementation
+
+The network was proposed in 'SuperGlue: Learning Feature Matching with Graph Neural Networks' and is implemented by 
+wrapping over author's source-code.
+
+Note: the pretrained model only supports SuperPoint detections currently.
+
+References:
+- http://openaccess.thecvf.com/content_CVPR_2020/papers/Sarlin_SuperGlue_Learning_Feature_Matching_With_Graph_Neural_Networks_CVPR_2020_paper.pdf # noqa
+- https://github.com/magicleap/SuperGluePretrainedNetwork
+
+Authors: Ayush Baid, John Lambert
+"""
+from typing import Tuple
+
+import numpy as np
+import torch
+
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.frontend.matcher.matcher_base import MatcherBase
+from thirdparty.SuperGluePretrainedNetwork.models.superglue import SuperGlue
+
+SUPERGLUE_DESC_DIM = 256
```

Comment:
another yaml parameter?

Reply:
so this one is actually rigidly fixed, because we are using their pre-trained weights, which are set to 256-dimensional. So it's more an implementation detail on the 3rd party end, rather than something we can change. I put it up here for clarity, so we didn't have a random magic number "256" laying around.